### PR TITLE
WIP: Add compression to API

### PR DESCRIPTION
**Pull request by wussler**
- Add `EncryptWithCompression` to encrypt specifying a compression for asymmetric and session keys
```go
(keyRing *KeyRing) EncryptWithCompression(
	message *PlainMessage,
	privateKey *KeyRing,
	compressionAlgorithm packet.CompressionAlgo,
	level int,
) (*PGPMessage, error)

(sk *SessionKey) EncryptWithCompression(
	message *PlainMessage,
	compressionAlgorithm packet.CompressionAlgo,
	level int,
) ([]byte, error)
```

- Add tests (still broken)

### CODE REVIEW

**Comment by twiss** [comment]
Just for what it's worth - there isn't really any reason to pass the compression algorithm here in my opinion, since ZIP and ZLIB (the only supported algorithms for compression in ProtonMail/crypto) are for all intents and purposes the same (just the header is different).
Then, this is a bit more debatable but I also wouldn't add a parameter for the compression level. It's a quite low-level parameter, that already in ProtonMail/crypto usually shouldn't be necessary to pass I feel like, and in a higher level library like gopenpgp I would just abstract it away. If at some point we find out that a higher compression level is very beneficial, then I would rather change it here than at the call site.

**Comment by wussler** [comment]
> Then, this is a bit more debatable but I also wouldn't add a parameter for the compression level. It's a quite low-level parameter, that already in ProtonMail/crypto usually shouldn't be necessary to pass I feel like, and in a higher level library like gopenpgp I would just abstract it away. If at some point we find out that a higher compression level is very beneficial, then I would rather change it here than at the call site.

Yeah, probably it's better from a (simple) interface perspective to remove those values :)

**Comment by chesnokovilya** [comment]
This PR is dependent on https://github.com/ProtonMail/crypto/pull/62.

**Review by chesnokovilya — APPROVED** [review]

File: constants/cipher.go [hunk 1]
```diff
@@ -16,3 +16,6 @@ const (
 	SIGNATURE_NO_VERIFIER int = 2
 	SIGNATURE_FAILED      int = 3
 )
+
+const DefaultCompression = 2 // ZLIB
+const DefaultCompressionLevel = 6
```
[the remark this answers is not in the document]
Comment (wussler) [reply constants/cipher.go:21]: Good idea, I'll add the comment

**Review by twiss — APPROVED** [review]

**Pull request merged**